packages/server/lib/routes.ts

```diff
@@ -47,6 +47,74 @@ export const createRoutes = ({
     runner.handle(req, res)
   })
 
+  app.get('/__cypress/automation/getLocalStorage', (req, res) => {
+    res.send(`<html><body><script>(${(function () {
+      const _localStorageStr = JSON.stringify(window.localStorage)
+      const _localStorage = _localStorageStr.length > 2 && JSON.parse(_localStorageStr)
+      const _sessionStorageStr = JSON.stringify(window.sessionStorage)
+      const _sessionStorage = _sessionStorageStr.length > 2 && JSON.parse(JSON.stringify(window.sessionStorage))
+
+      const value = {} as any
+
+      if (_localStorage) {
+        value.localStorage = _localStorage
+      }
+
+      if (_sessionStorage) {
+        value.sessionStorage = _sessionStorage
+      }
```

Comment:
perhaps this would be good to deduplicate with the earlier usage of this pattern, or at least comment it, since at first glance it appears to be the work of a madperson

Reply:
we cant really dedupe here since this function is stringified and evalled in cross origin iframes. added comment